src/Servers/Kestrel/Core/src/Systemd/KestrelServerOptionsSystemdExtensions.cs

```diff
@@ -36,7 +37,17 @@ public static KestrelServerOptions UseSystemd(this KestrelServerOptions options,
         {
             if (string.Equals(Process.GetCurrentProcess().Id.ToString(CultureInfo.InvariantCulture), Environment.GetEnvironmentVariable(ListenPidEnvVar), StringComparison.Ordinal))
             {
-                options.ListenHandle(SdListenFdsStart, configure);
+                if (!byte.TryParse(Environment.GetEnvironmentVariable(ListenFdsEnvVar), NumberStyles.None, NumberFormatInfo.InvariantInfo, out var listenFds))
```

Comment:
`byte`. Is it documented anywhere that LISTEN_FDS cannot be larger than 255?

Reply:
No, this is actually [allowed to be in range of [1;INT_MAX]](https://github.com/systemd/systemd/blob/0f2d351f790516bc3f1158d964c5f83f339addb2/src/libsystemd/sd-daemon/sd-daemon.c#L73), but I thought it might be a good idea to limit it to a more reasonable number since I don't really believe that supporting inheritance of INT_MAX-1 sockets is a practical idea because I don't think it would be robust and/or testable.

Reply:
Still, matching systemd's "native" functionality is not an issue to me, if you believe this is a correct way of handling this.

Reply:
I also opted out of hard failure when invalid values for `%LISTEN_FDS%` (like `0` or `-84`) are presented, not sure if this is a proper course of action, since systemd "native" library is very explicit about invalid values (considers this an error) and missing envvar (treats this as a non-socket-activated case).

Reply:
Sorry for the slow reply, I think we should change this to an `int`. If LISTEN_FDS is really large, so be it.

Reply:
No problems, Stephen, had a lot on my plate recently too. I'll match `sd_listen_fds` behavior then.

Reply:
I think I fixed this, also rebased to more recent head.